src/main/java/org/truffleruby/platform/darwin/DarwinPlatform.java

```diff
@@ -117,9 +101,10 @@ public FDSet createFDSet() {
     }
 
     @Override
-    public long createSigAction(long handler) {
-        long structSigAction = nfi.allocate(16); // sizeof(struct sigaction)
-        nfi.putLong(structSigAction + 0, handler); // offsetof(struct sigaction, sa_handler)
+    public Pointer createSigAction(long handler) {
+        Pointer structSigAction = Pointer.malloc(16); // sizeof(struct sigaction)
+        structSigAction.writeBytes(0, 16, (byte) 0);
```

Comment:
We should have another factory method to allocate + zero-out memory.
Maybe `Pointer.calloc(16)`?